server/src-lib/Hasura/Tracing.hs

```diff
@@ -213,19 +233,26 @@ extractHttpContext :: [HTTP.Header] -> IO (Maybe TraceContext)
 extractHttpContext hdrs = do
   freshSpanId <- liftIO Rand.randomIO
   pure $ TraceContext
-    <$> (HTTP.parseHeaderMaybe =<< lookup "X-Hasura-TraceId" hdrs)
+    <$> (fmap hexToWord64 =<< HTTP.parseHeaderMaybe =<< lookup "X-B3-TraceId" hdrs)
     <*> pure freshSpanId
-    <*> pure (HTTP.parseHeaderMaybe =<< lookup "X-Hasura-SpanId" hdrs)
+    <*> pure (fmap hexToWord64 =<< HTTP.parseHeaderMaybe =<< lookup "X-B3-SpanId" hdrs)
+
 
 -- | Inject the trace context as a JSON value, appropriate for 
 -- storing in (e.g.) an event trigger payload.
 injectEventContext :: TraceContext -> J.Value
-injectEventContext ctx =
-  J.object
-    [ "trace_id" J..= tcCurrentTrace ctx
-    , "span_id"  J..= tcCurrentSpan ctx
-    ]
-
+injectEventContext TraceContext{..} =
+  case tcCurrentParent  of
+    Nothing -> J.object
+                [ "trace_id" J..= tcCurrentTrace 
+                , "span_id"  J..= tcCurrentSpan 
+                ]
+    Just parentID -> J.object
+                      [ "trace_id" J..=  tcCurrentTrace 
+                      , "span_id"  J..=  tcCurrentSpan 
+                      , "parent_id" J..=  parentID
```

Comment:
We encode the parent ID here but never read it. I don't think it's necessary.

Reply:
Fixed it :)